src/test/scala/DSATest.scala

```diff
@@ -0,0 +1,707 @@
+    var program = prog(
+      proc("main",
+        block("operations",
+          locAssign1, // R6 = R0 + 4
+          locAssign2, // R7 = R0 + 5
+          MemoryAssign(mem,  R7, R1, BigEndian, 64, Some("00003")),
+          MemoryAssign(mem,  R6, R2, BigEndian, 64, Some("00004")),
+          ret
+        )
+      )
+    )
+
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    program = returnUnifier.visitProgram(program)
+    val results = RunUtils.staticAnalysis(StaticAnalysisConfig(None, None, None), IRContext(Set.empty, Set.empty, Set.empty, Map.empty, Specification(Set(), Set(), Map(), List(), List(), List(), Set()), program))
+    val dsg: DSG = results.locals.get(program.mainProcedure)
+
+    // object of formals R1 and R2 are written to overlapping fields of the same node? causing them to be merged together
+    assert(dsg.adjust(dsg.formals(R1)).equals(dsg.adjust(dsg.formals(R2))))
+
+    // R6 and R7 address the same cell
+    assert(dsg.find(dsg.varToCell(locAssign1)(R6)).cell.equals(dsg.find(dsg.varToCell(locAssign2)(R7)).cell))
+    // however, they address different internal offets in those cells
+    assert(dsg.find(dsg.varToCell(locAssign1)(R6)).internalOffset == 0)
+    assert(dsg.find(dsg.varToCell(locAssign2)(R7)).internalOffset == 1)
+    // R6 points to input R1
+    assert(dsg.adjust(dsg.varToCell(locAssign1)(R6))._pointee.isDefined)
+    assert(dsg.adjust(dsg.adjust(dsg.varToCell(locAssign1)(R6)).getPointee).equals(dsg.adjust(dsg.formals(R1))))
+
+  }
+
+  test("offsetting from middle of cell to a new cell") {
+    val mem = SharedMemory("mem", 10000, 10000)
+    val locAssign1 = Assign(R6, BinaryExpr(BVADD, R0, BitVecLiteral(4, 64)), Some("00001"))
+    val locAssign2 = Assign(R7, BinaryExpr(BVADD, R0, BitVecLiteral(5, 64)), Some("00002"))
+    val locAssign3 = Assign(R5, BinaryExpr(BVADD, R7,  BitVecLiteral(8, 64)), Some("00005"))
+
+    var program = prog(
+      proc("main",
+        block("operations",
+          locAssign1, // R6 = R0 + 4
+          locAssign2, // R7 = R0 + 5
+          MemoryAssign(mem, R7, R1, BigEndian, 64, Some("00003")),
+          MemoryAssign(mem, R6, R2, BigEndian, 64, Some("00004")),
+          locAssign3, // R5 = R7 + 8
+          ret
+        )
+      )
+    )
+
+    val returnUnifier = ConvertToSingleProcedureReturn()
+    program = returnUnifier.visitProgram(program)
+
+    val results = RunUtils.staticAnalysis(StaticAnalysisConfig(None, None, None), IRContext(Set.empty, Set.empty, Set.empty, Map.empty, Specification(Set(), Set(), Map(), List(), List(), List(), Set()), program))
+    val dsg: DSG = results.locals.get(program.mainProcedure)
+    // check that R5 points to separate cell at offset 13
+    assert(dsg.find(dsg.varToCell(locAssign3)(R5)).offset == 13)
+  }
+
+  test("offsetting from middle of cell to the same cell") {
+    // similar to above except instead of creating new cell the last assign
+    // points R7's cell at an internal offset of 8
```

Comment:
This isn't what the test actually tests, it tests that R7 has an internal offset of 1 and that R5 has an internal offset of 8.